### DISCUSSION TITLE
How to correctly persist store data inside local storage(NextJS)

### DISCUSSION THREAD

**Discussion by victornasar** [pmndrs/zustand#2230] [Q&A] [answered]
My first time using Zustand. I've been trying to persist my cart and stripe product when I refresh the my nextjs dynamic page. So far, I am able to persist on the client side, but the server side always returns as undefined. When I try to render product information on my page, it returns an error which stems from my product in useCart being undefined.

<img width="520" alt="Screenshot 2023-12-06 at 12 11 55 PM" src="https://github.com/pmndrs/zustand/assets/44656132/b895d8bd-94d7-4d79-a316-cac0d58a98a1">
<img width="520" alt="Screenshot 2023-12-06 at 12 12 23 PM" src="https://github.com/pmndrs/zustand/assets/44656132/80f1282b-7b7d-4bcb-b8cf-993f1d02dec8">
 
<img width="615" alt="Screenshot 2023-12-06 at 12 13 10 PM" src="https://github.com/pmndrs/zustand/assets/44656132/7deee48f-c515-4a0b-9499-a9f3ea53815b">

**Comment by dbritto-dev** [comment]
@victornasar please read this first -> https://docs.pmnd.rs/zustand/integrations/persisting-store-data#usage-in-next.js

**Reply by victornasar** [reply, marked as answer]
I just solved it, it was a spelling error this whole time.
My apologies
